dart/tool/dart/generate_dart.dart

```diff
@@ -916,6 +916,26 @@ class Type extends Member {
 
   Type(this.parent, String categoryName, String definition, [this.docs]) {
     _parse(new Tokenizer(definition).tokenize());
+    // Anything with an `id` should have a `fixedId` boolean also, but that
+    // isn't in the protocol definition.
+    if (fields.any((f) => f.name == 'id') &&
+        !fields.any((f) => f.name == 'fixedId')) {
+      var field = TypeField(this, '')
+        ..type = (MemberType()..types = [TypeRef('bool')])
+        ..name = 'fixedId'
+        ..optional = true;
+      fields.add(field);
+    }
+
+    // The vm sends a name for VM objects even though that isn't in the
+    // protocol.
+    if (name == 'VM' && !fields.any((f) => f.name == 'name')) {
+      var field = TypeField(this, '')
```

Comment:
@jakemac53, you may consider patching the spec in the repo (w/ a comment), instead of special-casing this here. We could then upstream the same change to the sdk.

Reply:
I think for now this is a bit easier to maintain than an edited spec? It is written in a way such that it will be a no-op if the spec does get updated.

Happy to follow up once we hear more though, hopefully we can just update these things upstream.

Reply:
The `name` property should always be set for `@VM` and `VM`.